src/idl_parser.cpp

```diff
@@ -191,7 +191,7 @@ std::string Parser::TokenToStringId(int t) {
 // Parses exactly nibbles worth of hex digits into a number, or error.
 CheckedError Parser::ParseHexNum(int nibbles, int64_t *val) {
   for (int i = 0; i < nibbles; i++)
-    if (!isxdigit(cursor_[i]))
+    if (!isxdigit(static_cast<const unsigned char>(cursor_[i])))
       return Error("escape code must be followed by " + NumToString(nibbles) +
```

Comment:
This is already being done down on line 353:
`while (isxdigit(static_cast<unsigned char>(*cursor_))) cursor_++;`
The exact warning is:
`idl_parser.cpp(194): warning : C6330: 'const char' passed as _Param_(1) when 'unsigned char' is required in call to 'isxdigit'.`